### PR TITLE
IBP-6088-implement-table-as-strategy-for-sequence-generation-in-attribute-germplasm-germplasm-study-source-and-name-domain-models

### PR DESCRIPTION
**Pull request by lik-83**
…ibutes, germplasm, names and germplasm study source domain models

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/pojos/Germplasm.java [hunk 1]
```diff
@@ -240,6 +244,7 @@ public class Germplasm extends AbstractEntity implements Serializable, Cloneable
 	private Integer mgid;
 
 	@OneToMany(mappedBy = "germplasm", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
+	@BatchSize(size = 5000)
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/pojos/Germplasm.java:247]: @lik-83  is this batch size intentional?

Comment (lik-83) [reply src/main/java/org/generationcp/middleware/pojos/Germplasm.java]: It was not necessary, so I removed it.

File: src/main/resources/liquibase/crop_changelog/TBD.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    </changeSet>
+
+	<changeSet author="lkovacic" id="vTBD-4">
+        <preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="0">
+				select count(1) from sequence where sequence_name = 'names'
+			</sqlCheck>
+        </preConditions>
+        <comment>
+            Create sequence entry for germplasm_study_source table generator
```
Comment (clarysabel) [inline src/main/resources/liquibase/crop_changelog/TBD.xml:59]: @lik-83 small detail to fix comment on this changeset

Comment (lik-83) [reply src/main/resources/liquibase/crop_changelog/TBD.xml]: Done!

File: src/main/resources/liquibase/crop_master.xml [hunk 3]
```diff
@@ -99,6 +99,7 @@
 	<include file="crop_changelog/22_2_0.xml" relativeToChangelogFile="true"/>
 	<include file="crop_changelog/22_3_0.xml" relativeToChangelogFile="true"/>
 	<include file="crop_changelog/22_4_0.xml" relativeToChangelogFile="true"/>
+	<include file="crop_changelog/TBD.xml" relativeToChangelogFile="true"/>
```
[the remark this answers is not in the document]
Comment (lik-83) [reply src/main/resources/liquibase/crop_master.xml]: Done!

**Pull request merged**